src/telev/telev.c

```diff
@@ -275,15 +296,15 @@ int telev_poll(struct telev *tel, bool *marker, struct mbuf *mb)
 		mrk = true;
 
 		tel->event = mbuf_read_u8(tel->mb);
-		tel->dur   = tel->ptime * 8;
+		tel->dur   = tel->ptime * (tel->srate / 1000);
 		tel->state = SENDING;
 		tel->txc   = 1;
 
 		err = payload_encode(mb, tel->event, false, tel->dur);
 		break;
 
 	case SENDING:
-		tel->dur += tel->ptime * 8;
+		tel->dur += tel->ptime * (tel->srate / 1000);
```

Comment:
this could be optimized.

if the value of ptime * srate / 1000 is calculated in the alloc function and saved,
it can be used instead. Thus avoiding a multiplication and division for every packet sent.